lib/timezone/lookup/geonames.rb

```diff
@@ -34,13 +34,37 @@ def lookup(lat, long)
         raise(Timezone::Error::GeoNames, e.message)
       end
 
+      def lookup_with_etc(lat, long)
+        response = client.get(url(lat, long))
+
+        return unless response.body
+
+        data = JSON.parse(response.body)
+
+        return data['timezoneId'] if data['timezoneId']
+
+        return get_etc(data['gmtOffset']) if data['gmtOffset']
+
+        return unless data['status']
+
+        raise(Timezone::Error::GeoNames, data['status']['message'])
+      rescue => e
+        raise(Timezone::Error::GeoNames, e.message)
+      end
+
       private
 
+      def get_etc(offset)
+        return 'Etc/UTC' if offset == 0
+        "Etc/GMT#{format('%+d', offset.to_i)}"
```

Comment:
You don't need to worry about compatibility in this case. `lookup(0,0)` may raise an exception today and work tomorrow even if there is no gem change - simply because the API results change. The API is still the same, so you aren't breaking backwards compatibility.

Thinking about this further, my only concern is that post-processing may be confusing because this code is trying to make a best guess for what a timezone would be if a land-form existed there. For example, while it is reasonable to expect that the middle of the Pacific Ocean has an `Etc/UTC+/-..` timezone, it is also not really correct because there is no official timezone there - because there is no land there.

I would vote for adding a configuration option to `Timezone::Lookup.config(:geonames)` like:

```
Timezone::Lookup.config(:geonames) do |c|
  c.offset_etc_zones = true
end
```

The default value would be `false`.

You can then use that configuration option in the `#lookup` function to include or exclude responses w/o a timezone id.

Does this make sense?